File: VECTOAux/VectoAuxiliaries/Electrics/CombinedAlternatorMapRow.vb
Imports DownstreamModules.Electrics

Namespace Electrics
	'This class is reflective of the stored entries for the combined alternator
	'And is used by the Combined Alternator Form and any related classes.

	Public Class CombinedAlternatorMapRow
		Implements ICombinedAlternatorMapRow

		Public Property AlternatorName As String Implements ICombinedAlternatorMapRow.AlternatorName
		Public Property RPM As Double Implements ICombinedAlternatorMapRow.RPM
		Public Property Amps As Double Implements ICombinedAlternatorMapRow.Amps
		Public Property Efficiency As Double Implements ICombinedAlternatorMapRow.Efficiency
		Public Property PulleyRatio As Double Implements ICombinedAlternatorMapRow.PulleyRatio

		'Constructors
		Sub New()
		End Sub

		Sub New(AlternatorName As String, RPM As Single, Amps As Single, Efficiency As Single, PulleyRatio As Single)

			'Sanity Check
			If AlternatorName.Trim.Length = 0 Then Throw New ArgumentException("Alternator name cannot be zero length")
			If Efficiency < 0 Or Efficiency > 100 Then _
				Throw New ArgumentException("Alternator Efficiency must be between 0 and 100")
			If PulleyRatio <= 0 Then Throw New ArgumentException("Alternator Pully ratio must be a positive number")

			'Assignments
			Me.AlternatorName = AlternatorName
			Me.RPM = RPM
			Me.Amps = Amps
			Me.Efficiency = Efficiency
			Me.PulleyRatio = PulleyRatio
		End Sub
	End Class
End Namespace

File: VECTOAux/BusAuxiliaries/DownstreamModules/Impl/Electrics/CombinedAlternatorMapRow.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using DownstreamModules.Electrics;

namespace Electrics
{
	// This class is reflective of the stored entries for the combined alternator
	// And is used by the Combined Alternator Form and any related classes.

	public class CombinedAlternatorMapRow : ICombinedAlternatorMapRow
	{
		public string AlternatorName { get; set; }
		public double RPM { get; set; }
		public double Amps { get; set; }
		public double Efficiency { get; set; }
		public double PulleyRatio { get; set; }

		// Constructors
		public CombinedAlternatorMapRow()
		{
		}

		public CombinedAlternatorMapRow(string alternatorName, double rpm, double amps, double efficiency, double pulleyRatio)
		{

			// Sanity Check
			if (alternatorName.Trim().Length == 0)
				throw new ArgumentException("Alternator name cannot be zero length");
			if (efficiency < 0 | efficiency > 100)
				throw new ArgumentException("Alternator Efficiency must be between 0 and 100");
			if (pulleyRatio <= 0)
				throw new ArgumentException("Alternator Pully ratio must be a positive number");

			// Assignments
			AlternatorName = alternatorName;
			RPM = rpm;
			Amps = amps;
			Efficiency = efficiency;
			PulleyRatio = pulleyRatio;
		}
	}
}
